File: 01/js/bubble.js
const TestHelper = require('./testHelper.js');
const Algorithm = require('./algorithm.js');

const n = 10000;
const arr = TestHelper.generateRandomArray(n, 0, n);
// const arr = TestHelper.generateNearlyOrderedArray(n, 10);
const arr2 = TestHelper.copyArray(arr);
const arr3 = TestHelper.copyArray(arr);

TestHelper.testSort('   bubble sort', Algorithm.bubbleSort, arr);
TestHelper.testSort('insertion sort', Algorithm.insertionSort, arr2);
TestHelper.testSort('selection sort', Algorithm.selectionSort, arr3);

File: 01/cpp/bubble.cpp
#include <iostream>
#include "testHelper.h"
#include "algorithm.h"

using namespace std;

int main() {
	int n = 10000;
	int *arr = TestHelper::generateRandomArray(n, 0, n);
	// int *arr = TestHelper::generateNearlyOrderedArray(n, 10);
	int *arr2 = TestHelper::copyIntArray(arr, n);
	int *arr3 = TestHelper::copyIntArray(arr, n);

	TestHelper::testSort("   bubble sort", Algorithm::bubbleSort, arr, n);
	TestHelper::testSort("insertion sort", Algorithm::insertionSort, arr2, n);
	TestHelper::testSort("selection sort", Algorithm::selectionSort, arr3, n);

	delete[] arr;
	delete[] arr2;

	return 0;
}